Docs update

ShahFragcolor opened the pull request:
<!--
Before submitting your PR, please review the following checklist:

- [ ] **DO NOT** submit a PR without having discussed the change first in a related issue. If none exist, create one first.
- [ ] **CONSIDER** adding a unit test if your PR resolves an issue.
- [ ] **DO** keep pull requests small so they can be easily reviewed.
- [ ] **DO** make sure all unit tests pass.
- [ ] **DO** make sure not to introduce any new compiler warnings.
- [ ] **AVOID** breaking the continuous integration build.
- [ ] **AVOID** making significant changes to the overall architecture.
-->

ShahFragcolor committed:
update-1

ShahFragcolor committed:
update-2

ShahFragcolor committed:
finished shards flow doc

sinkingsugar commented on docs/docs/learn/shards/primer/coding-with-shards.md:
````diff
@@ -154,27 +157,26 @@ Imagine a scenario where you have a float `3.141592653589793` that you need to r
-    .pi-value (Math.Add .pi-value) (Math.Multiply .pi-value) (Math.Subtract .pi-value)
+    ```shards
+    3.141592653589793 = pi-value ;; (1)
+    pi-value | Math.Add(.pi-value) | Math.Multiply(pi-value) | Math.Subtract(pi-value)
````
```suggestion
    pi-value | Math.Add(pi-value) | Math.Multiply(pi-value) | Math.Subtract(pi-value)
```

sinkingsugar commented on docs/docs/learn/shards/primer/coding-with-shards.md:
````diff
@@ -198,145 +200,146 @@ Here are the variable types and the symbols used to create and assign to them:
-        ;; your shards here
+    ```shards
+
+    @defshards( message-groups {
````
whats this?

sinkingsugar commented on docs/docs/learn/shards/primer/coding-with-shards.md:
````diff
@@ -198,145 +200,146 @@ Here are the variable types and the symbols used to create and assign to them:
-    
+
+    ```shards
+    (send-message "Hello World!")
````
?

sinkingsugar commented on docs/docs/learn/shards/primer/coding-with-shards.md:
````diff
@@ -198,145 +200,146 @@ Here are the variable types and the symbols used to create and assign to them:
+
+=== "Code"
+
     ```{.clojure .annotate linenums="1"}
````
shards

sinkingsugar commented on docs/docs/learn/shards/primer/coding-with-shards.md:
````diff
@@ -198,145 +200,146 @@ Here are the variable types and the symbols used to create and assign to them:
-    (send-message "Hello World!")
+    ```shards
+
+    send-message("Hello World!")
````
```suggestion
    @send-message("Hello World!")
```
maybe?

sinkingsugar commented on docs/docs/learn/shards/primer/coding-with-shards.md:
````diff
@@ -198,145 +200,146 @@ Here are the variable types and the symbols used to create and assign to them:
+    ```shards
+
+    Repeat(
+     Action:{
````
```suggestion
    Repeat(
     Action: {
```

In general formatting should be consistent

sinkingsugar commented on docs/docs/learn/shards/primer/coding-with-shards.md:
````diff
@@ -198,145 +200,146 @@ Here are the variable types and the symbols used to create and assign to them:
 
 === "Code"
     
     ```{.clojure .annotate linenums="1"}
````
shards

sinkingsugar commented on docs/docs/learn/shards/primer/coding-with-shards.md:
```diff
@@ -198,145 +200,146 @@ Here are the variable types and the symbols used to create and assign to them:
-    (Repeat
-     :Action (msg-one-to-five)
-     :Times 5)
+    @define(msg-one-to-five {
```
I don't understand those examples, I think they were not great to begin with

sinkingsugar reviewed (changes requested):
Gave it a review pass, in general this page was made a long time ago pre-ai... we should instead feed it to claude + shards knowledge doc to make it better ideally

ShahFragcolor committed:
update-3

ShahFragcolor committed:
converted lisp functions to built in functions

sinkingsugar commented on docs/docs/reference/shards/built-ins/index.md:
```diff
@@ -0,0 +1,20 @@
+---
```
where are `@template` and `@macro` and `@define`? I saw somewhere else but they should be here too no?

sinkingsugar reviewed (commented):
So far so good but I don't see some stuff

ShahFragcolor committed:
built-in section rework

- added @define @template and @macro

- reorganised how the built ins are clasified

ShahFragcolor committed:
Updated Reference page Scripts

- updated Types under reference to use correct syntax

- updated GFX under reference to use correct syntax

- switched "Rare Engine" to Formabble Engine

- Removed Download and Contribute Section

sinkingsugar commented on shards/tests/temp-test.shs:
```diff
@@ -0,0 +1,8 @@
+@mesh(main)
```
Remove this I guess, can add to `.gitignore` as well

ShahFragcolor committed:
Reworked example

- reworked the coding with shards examples

- deleted temp-test.sh which i accidentally left in

ShahFragcolor committed:
- added compile time evaluation and parantheses control

- added segments about compile time evaluation and paranthesis control
- changed compile time terminology to compose time

- added some terminology from shards-generic.md from edgetalk

sinkingsugar commented on docs/docs/learn/shards/primer/coding-with-shards.md:
````diff
@@ -4,27 +4,27 @@ In this chapter, we will be learning how to code with Shards so that you can wri
 
-![Some shard examples.](assets/shards-examples.png)
+```shards
+Math.Add(Operand: 1)
````
should showcase that parentheses are optional actually as well

sinkingsugar commented on docs/docs/learn/shards/primer/coding-with-shards.md:
```diff
@@ -4,27 +4,27 @@ In this chapter, we will be learning how to code with Shards so that you can wri
 ![The parameters set for a shard affects it's behavior.](assets/shard-parameter.png)
 
-In code form, parameters are defined by the user within the parentheses of the shard itself, after the shard's name. The above examples will appear as `5 (Math.Add 1)` and `5 (Math.Add 3)` in code. 
+In code form, parameters are defined by the user within the parentheses after the shard's name. The above examples will appear as `5 Math.Add(Operand: 1)` and `5 Math.Add(3)` in code. 
```
probably add `|`, altho they are optional (and we should mention somewhere)

sinkingsugar commented on docs/docs/learn/shards/primer/coding-with-shards.md:
````diff
@@ -198,145 +200,185 @@ Here are the variable types and the symbols used to create and assign to them:
     ```
 
-Let us now take a look at how we can utilize `defshards` in a code snippet that counts from 1 to 5 multiple times.
+Let us now take a look at how we can utilize `@define`. Let's say we have a player that can be damaged by different sources.
````
Don't we have a dedicated section for those? Maybe we should just give a tldr and link there?

sinkingsugar commented on docs/docs/reference/shards/built-ins/data-construction.md:
```diff
@@ -8,7 +8,7 @@ license: CC-BY-SA-4.0
 
 * These built-in functions construct integer vector types (int2, int3, int4, int8, int16).
-* If they are called in expression position and have only constants for parameters, the value is evaluated at compile time.
+* If they are called in expression position and have only constants for parameters, the value is evaluated at compose time.
```
whats expression position?

sinkingsugar commented on docs/docs/reference/shards/built-ins/data-construction.md:
````diff
@@ -33,7 +33,7 @@ Include `@i2`, `@i3`, `@i4`, `@i8`, `@i16`
     
         ```shards
-        Math.Add(Operand: @i2(1 1)) ;; compile time evaluation
+        Math.Add(Operand: @i2(1 1)) ;; compose time evaluation
````
not sure what this means, there is no compose evaluation, it's just a constant i2?

sinkingsugar commented on docs/docs/learn/shards/primer/coding-with-shards.md:
````diff
@@ -445,6 +445,19 @@ When the Mesh is run, the Wires are executed in sequence and your program is sta
         @run(mesh-name (1.0 | Math.Divide(60.0)))
         ```
 
+!!! note "Parentheses to Control Evaluation Order"
````
This is important, we probably want a section above, rather than just a note as it's a language construct

ShahFragcolor committed:
- coding with shards fixes and shards lifecycle

- parentheses and # has its own section now
- added a note about | syntax sugar
- made @template and @define shorter by removing the generic syntax part and linking them to built ins
- removed erroneous segment about expression position in data-construction
- added temp-tests.shs to git-ignore
